components/raftstore/src/store/peer.rs

```diff
@@ -2322,6 +2336,16 @@ where
                             "region_id" => self.region_id,
                         );
                     }
+                    // After the leadership changed, send `CasualMessage::Campaign` to the target
+                    // peer to campaign leader if there exists uncampaigned regions. It's used to
+                    // ensure that a leader is elected promptly for the newly created Raft
+                    // group, minimizing availability impact (e.g. #12410 and #17602.).
+                    for new_region in self.uncampaigned_new_regions.0.drain(..) {
+                        let _ = ctx.router.send(
+                            new_region,
+                            PeerMsg::CasualMessage(Box::new(CasualMessage::Campaign)),
```

Comment:
Sending a `CasualMessage::Campaign` here may break linearizability, I will think about this more.

Reply:
Are there any specific corner cases or examples that we should consider?

Reply:
It's okay, as long as the campaign happens after parent region elects a leader